src/config/file/v1.rs

```diff
@@ -135,23 +148,46 @@ pub(super) enum Integration {
         title: Option<String>,
         remote: String,
         auth: Auth,
+        import_branches: Option<bool>,
+        import_tags: Option<bool>,
+        watched_branches: Option<Vec<String>>,
     },
 }
 
 impl TryFrom<Integration> for remote::Integration {
     type Error = Report<remote::ValidationError>;
 
     fn try_from(value: Integration) -> Result<Self, Self::Error> {
-        match value {
+        let mut integration = match value {
             Integration::Git {
                 poll_interval,
                 remote,
                 team,
                 title,
                 auth,
+                import_branches,
+                import_tags,
+                watched_branches,
             } => {
                 let poll_interval = remote::PollInterval::try_from(poll_interval)?;
                 let endpoint = remote::Remote::try_from(remote)?;
+                let import_branches = remote::BranchImportStrategy::from(import_branches);
+                let import_tags = remote::TagImportStrategy::from(import_tags);
+                let watched_branches = watched_branches
+                    .unwrap_or_default()
+                    .into_iter()
+                    .map(remote::WatchedBranch::new)
+                    .collect::<Vec<_>>();
+
+                if import_branches == remote::BranchImportStrategy::Disabled
+                    && !watched_branches.is_empty()
+                {
```

Comment:
In the selected code, the validation is happening on the Disabled variant. I restructured the implementation to look like this:

```
impl BranchImportStrategy {
    /// Validates branch import configurations
    pub fn is_valid(&self, watched_branches: &[WatchedBranch]) -> bool {
        match self {
            BranchImportStrategy::Disabled => watched_branches.is_empty(),
            // On the Enabled variant, we can't check if watched branches is not empty here because Broker will try to infer watched branches
            // Therefore just have the Enabled case map to true, and the validation on the enabled case will be checked later
            BranchImportStrategy::Enabled => true,
        }
    }

    /// Checks if we need to infer watched branches based on configuration
    pub fn infer_watched_branches(&self, watched_branches: &[WatchedBranch]) -> bool {
        match self {
            BranchImportStrategy::Disabled => false,
            BranchImportStrategy::Enabled => watched_branches.is_empty(),
        }
    }
```